### PR TITLE
Sometimes toast about old join community notification starts appearing on every login

### LINKED ISSUE #17955: Sometimes toast about old join community notification starts appearing on every login
There are no reproduction steps yet. Hope attached logs might help. 

For now I have faced the issue only on Android but I don't think it is device specific. It is hard to verify as it happens randomly.

On the video below (recorded on release build) you can see the toast appearing on every relogin. 

[Status-debug-logs (83).zip](https://github.com/status-im/status-mobile/files/13426193/Status-debug-logs.83.zip)

https://github.com/status-im/status-mobile/assets/97245802/bd716afe-e0fe-4bab-80d9-addf26bf1587

### Additional Information

- Status version: release 1.26, nightly
- Operating System: Android

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #17955 

### Summary
Fixes an issue that caused the 'you joined' community toast show after login.

status: ready

### CODE REVIEW

**Comment by ibrkhalil** [comment]
Already approved before https://github.com/status-im/status-mobile/pull/19460

**Comment by pavloburykh** [comment]
@ibrkhalil thank for the PR. So far I was not able to reproduce the issue. Ready for merge.

File: fastlane/Gemfile.lock [hunk 1]
```diff
@@ -217,6 +217,7 @@ GEM
 
 PLATFORMS
   arm64-darwin-22
+  x86_64-darwin-22
```
Comment (seanstrom) [inline fastlane/Gemfile.lock:220]: Is this change needed? This seems to be the only thing different from the previous PR, thoughts?

**Review by seanstrom — APPROVED** [review]
Nice stuff! ✅ 
I left a small comment about the Gemfile, might be worth double-checking.
cc @siddarthkay @jakubgs

File: fastlane/Gemfile.lock (hunk 1 above)
Comment (ibrkhalil) [reply fastlane/Gemfile.lock:220]: I left the changes there because I don't have an arm machine as I am using a Hackintosh and not Apple SIlicon and it was being changed everytime I ran make run-ios

**Comment by siddarthkay** [comment]
Seems weird, will take a look in a couple of hours, thanks for tagging

**Comment by clauxx** [comment]
@ibrkhalil would this be fixing the contact request notification issue I mentioned as well?

**Comment by ibrkhalil** [comment]
> @ibrkhalil would this be fixing the contact request notification issue I mentioned as well?

Nope, Will ask QA if there's a task for the contact request issue one.

File: fastlane/Gemfile.lock (hunk 1 above)
Comment (siddarthkay) [reply fastlane/Gemfile.lock:220]: Thanks for explaining, @seanstrom i believe you're on an arm machine? If so could you pls run `make run-ios` and see if this part remains unchanged. If so then this change is good to keep. Else we'd need to dig into it further...

Comment (seanstrom) [reply fastlane/Gemfile.lock:220]: I don't see that part change, but I have a `podfile.lock` change for some reason: https://github.com/status-im/status-mobile/commit/7bef8b0bade9fec768c1bec6281eae22d5773dbf

Comment (seanstrom) [reply fastlane/Gemfile.lock:220]: Just checked and that `podfile.lock` change seems to be present on develop, so perhaps unrelated to these changes.

**Comment by pavloburykh** [comment]
> @ibrkhalil would this be fixing the contact request notification issue I mentioned as well?

Hey @clauxx! We have not yet found steps on how to replicate the issue, though we are randomly facing it from time to time. Maybe you have any ideas on what are the steps? Thank you.

File: fastlane/Gemfile.lock (hunk 1 above)
Comment (seanstrom) [reply fastlane/Gemfile.lock:220]: Okay I triple checked by running `make clean` for develop and this branch, and then I ran `make run-ios` for each branch, and I see no diffs 🙌

**Comment by clauxx** [comment]
> > @ibrkhalil would this be fixing the contact request notification issue I mentioned as well?
> 
> Hey @clauxx! We have not yet found steps on how to replicate the issue, though we are randomly facing it from time to time. Maybe you have any ideas on what are the steps? Thank you.

not really. It just happens on a specific profile with one contact after I recover. But it does happen **every** time I recover this profile. You can try it with this seedphrase `type tissue number dune acid team today help daughter oval film just`

File: fastlane/Gemfile.lock (hunk 1 above)
Comment (siddarthkay) [reply fastlane/Gemfile.lock:220]: Thanks for verifying @seanstrom !!

**Comment by ibrkhalil** [comment]
> type tissue number dune acid team today help daughter oval film just

Reproduced it during onboarding when I restored that seed phrase, Will take a closer look soon.
Need to create a task for it.

**Comment by pavloburykh** [comment]
> not really. It just happens on a specific profile with one contact after I recover. But it does happen every time I recover this profile.

@clauxx thank you for sharing you account. 

> Reproduced it during onboarding when I restored that seed phrase, Will take a closer look soon. Need to create a task for it.

Actually, this is not the exact issue which we are looking for. In case of Cristian's account notification appears only once, after account is restored. It is not so critical, IMHO, but probably should be addressed as well. @ibrkhalil would you mind logging this issue? 

What we were facing before is contact request notification which is constantly appearing on every re-login. This case has higher priority and more anoying. But unfortunately we do not have reproduction steps yet.
